plugin/handlergen.go

```diff
@@ -72,6 +73,111 @@ func (p *OrmPlugin) generateReadHandler(message *generator.Descriptor) {
 	p.P()
 }
 
+func (p *OrmPlugin) generatePatchHandler(message *generator.Descriptor) {
+	var isMultiAccount bool
+	var hasIDField bool
+
+
+	typeName := p.TypeName(message)
+	ormable := p.getOrmable(typeName)
+
+	for _, field := range message.GetField() {
+		if strings.ToLower(field.GetName()) == "id" {
+			hasIDField = true
+			break
+		}
+	}
+
+	if opts := getMessageOptions(message); opts != nil && opts.GetMultiAccount() {
+		isMultiAccount = true
+	}
+
+	if isMultiAccount && !hasIDField {
+		p.P(fmt.Sprintf("// Cannot autogen DefaultPatch%s: this is a multi-account table without an \"id\" field in the message.\n", typeName))
+		return
+	}
+
+	p.P(`// DefaultPatch`, typeName, ` executes a basic gorm update call with patch behavior`)
+	p.P(`func DefaultPatch`, typeName, `(ctx context.Context, in *`,
+		typeName, `, updateMask *`, p.Import(fmImport), `.FieldMask, db *`, p.Import(gormImport), `.DB) (*`, typeName, `, error) {`)
+
+	p.P(`if in == nil {`)
+	p.P(`return nil, errors.New("Nil argument to DefaultPatch`, typeName, `")`)
+	p.P(`}`)
+	if isMultiAccount {
+		p.P("accountID, err := ", p.Import(authImport), ".GetAccountID(ctx, nil)")
+		p.P("if err != nil {")
+		p.P("return nil, err")
+		p.P("}")
+	}
+
+	// Read ORM object with preloaded subobjects. For simplicity we
+	// put the DefaultRead body here to have an ormObj to operate with
+	// in child association removals.
+
+	p.P(`ormParams, err := (&`, typeName, `{Id: in.GetId()}).ToORM(ctx)`)
+	p.P(`if err != nil {`)
+	p.P(`return nil, err`)
+	p.P(`}`)
+
+	if isMultiAccount {
+		p.P(`db = db.Where(&`, typeName, `ORM{AccountID: accountID})`)
+	}
+
+	p.generatePreloading(ormable)
+
+	p.P(`ormObj := `, ormable.Name, `{}`)
+	p.P(`if err := db.Where(&ormParams).First(&ormObj).Error; err != nil {`)
+	p.P(`return nil, err`)
+	p.P(`}`)
+
+	// Convert ormObject to protobuf to apply field mask.
+	// Note that due to synthetic fields support we cannot patch
+	// ormObj directly.
+	p.P(`pbObj, err := ormObj.ToPB(ctx)`)
+	p.P(`if err != nil {`)
+	p.P(`return nil, err`)
+	p.P(`}`)
+
+	// Patch pbObj with input according to a field mask.
+	p.P(`for _, f := range updateMask.GetPaths() {`)
+	for _, field := range message.GetField() {
```

Comment:
yes, I would like to not to use nested paths due to overcomplicated and not clear design (what to do with list values and so on), so if developer finds himself/herself willing to patch some value inside struct that is inside array i would suggest him/her to redesign resources and endpoints rather than doing such things.

Reply:
array accesses is explicitly not supported by the protobuf documentation, but nested paths are. https://godoc.org/google.golang.org/genproto/protobuf/field_mask

Reply:
so, the question is, whether we have to support a nested path or not?